src/item_factory.cpp

```diff
@@ -73,7 +73,8 @@ void Item_factory::finialize_item_blacklist()
         for( auto &elem : m_template_groups ) {
             elem.second->remove_item( itm );
         }
-        for( auto &recipes_b : recipes ) {
+        // FIXME: only recipes from by_category are erased;
+        for( auto &recipes_b : recipe_dict.by_category ) {
```

Comment:
Worry not!, the recipe deconstructor called a function that removes it from component lookup. It was kinda hidden that way so I thought it was. My bad for calling out the bug.